### PR TITLE
Prototype optional parameters

### PR DESCRIPTION
**Pull request by acotty**
I need to eventually add support for enabling exception breakpoints.
But in order to do this first I have to be able to see what features the existing DAP debugger supports.
But in order to do this I need to add support for the optional parameters.

I have added optional parameters to the wxEVT_DAP_INITIALIZE_RESPONSE event / dap::InitializeResponse structure for the supported boolean options. I have added optional parameters to a bunch of other structures that I will need to add support in the dap.cpp ::From(const Json& json) function when I get to them.

Before I go too much further I would liket o find out if I am heading in the right direction with this or should I be doing something completely differently? If I am heading in the right direction then do you think the changes will be acceptable or not and if not can you let me know what I need to do toy make them acceptable. If I am doing it wrong can you point me in the right direction I should be going.

### CODE REVIEW

File: dap/dap.hpp [hunk 1]
```diff
@@ -566,9 +569,211 @@ struct WXDLLIMPEXP_DAP InitializeRequest : public Request {
...
+};
+
+// Names of checksum algorithms that may be supported by a debug adapter.
+//
+// Must be one of the following enumeration values:
+// 'MD5', 'SHA1', 'SHA256', 'timestamp'
+using ChecksumAlgorithm = wxString;
+
+/// Information about the capabilities of a debug adapter.
+struct Capabilities {
```
Comment (eranif) [inline dap/dap.hpp]: all structs should be decorated with `WXDLLIMPEXP_DAP `

File: dap/dap.hpp [hunk 2]
```diff
@@ -566,9 +569,211 @@ struct WXDLLIMPEXP_DAP InitializeRequest : public Request {
...
+struct ColumnDescriptor {
+    // Name of the attribute rendered in this column.
+    wxString attributeName;
+
+    // Header UI label of column.
+    wxString label;
+
+    // Format to use for the rendered values in this column. TBD how the format
+    // strings looks like.
+    std::optional<wxString> format;
```
Comment (eranif) [inline dap/dap.hpp]: can't we use `wxAny` ? with value set to null to indicate "does not exist"

File: dap/dap.hpp (hunk 1 above)
Comment (acotty) [reply dap/dap.hpp]: Thanks. Will fix.

File: dap/dap.hpp (hunk 2 above)
Comment (acotty) [reply dap/dap.hpp]: I will check this out as I did not even look at wxAny. 

Thanks as this is exactly the feedback I was after.

File: dap/dap.hpp (hunk 1 above)
Comment (acotty) [reply dap/dap.hpp]: Just pushed the changes to fix this.

File: dap/dap.hpp (hunk 2 above)
Comment (acotty) [reply dap/dap.hpp]: Changes pushed to change to use wxAny.

**Pull request closed without merging**